### ISSUE TITLE
s3 sync issue on Linux

### ISSUE THREAD

**Issue by larntz** [laurent22/joplin#9549]
### Operating system

Linux

### Joplin version

2.13.9

### Desktop version info

Joplin 2.13.9 (prod, linux)

Client ID: b701f71ca2d544598d40ab51afb0d60a
Sync Version: 3
Profile Version: 44
Keychain Supported: No

Revision: 1bbec44

### Current behaviour

I use the s3 sync feature The same settings work on Android and MacOS X, but my Linux machine shows the error below. 

I am on Fedora 39 (amd64) using the Joplin AppImage. I have removed the sync settings and added them back. I've changed the secrete and access key to dummy values and got the same error. 

```
/tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/file-api-driver-amazon-s3.js:126 TypeError: Invalid character in header content ["authorization"]
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at new NodeError (node:internal/errors:399:5)
    at __node_internal_ (node:_http_outgoing:625:11)
    at ClientRequest.setHeader (node:_http_outgoing:648:3)
    at new ClientRequest (node:_http_client:292:14)
    at request (node:https:360:10)
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:72:25
    at new Promise (<anonymous>)
    at NodeHttpHandler.handle (/tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:50:16)
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/client-s3/dist-cjs/commands/DeleteObjectCommand.js:43:58
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:5:32
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:14:26
    at async /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-retry/dist-cjs/retryMiddleware.js:27:46
    at async /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26
(anonymous) @ /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/file-api-driver-amazon-s3.js:126
/tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/file-api-driver-amazon-s3.js:126 TypeError: Invalid character in header content ["authorization"]
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at new NodeError (node:internal/errors:399:5)
    at __node_internal_ (node:_http_outgoing:625:11)
    at ClientRequest.setHeader (node:_http_outgoing:648:3)
    at new ClientRequest (node:_http_client:292:14)
    at request (node:https:360:10)
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:72:25
    at new Promise (<anonymous>)
    at NodeHttpHandler.handle (/tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:50:16)
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/client-s3/dist-cjs/commands/DeleteObjectCommand.js:43:58
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:5:32
    at /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:14:26
    at async /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-retry/dist-cjs/retryMiddleware.js:27:46
    at async /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26
(anonymous) @ /tmp/.mount_JoplinspgUaR/resources/app.asar/node_modules/@joplin/lib/file-api-driver-amazon-s3.js:126
```

### Expected behaviour

Sync to S3.

### Logs

```
Logger.ts:254 14:44:36: Synchronizer: TypeError: Invalid character in header content ["authorization"]
    at __node_internal_captureLargerStackTrace (node:internal/errors:490:5)
    at new NodeError (node:internal/errors:399:5)
    at __node_internal_ (node:_http_outgoing:625:11)
    at ClientRequest.setHeader (node:_http_outgoing:648:3)
    at new ClientRequest (node:_http_client:292:14)
    at request (node:https:360:10)
    at /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:72:25
    at new Promise (<anonymous>)
    at NodeHttpHandler.handle (/tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/node-http-handler/dist-cjs/node-http-handler.js:50:16)
    at /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/client-s3/dist-cjs/commands/DeleteObjectCommand.js:43:58
    at /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-serde/dist-cjs/deserializerMiddleware.js:5:32
    at /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-signing/dist-cjs/middleware.js:14:26
    at async /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-retry/dist-cjs/retryMiddleware.js:27:46
    at async /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/node_modules/@aws-sdk/middleware-logger/dist-cjs/loggerMiddleware.js:7:26
log @ Logger.ts:254
warn @ Logger.ts:127
(anonymous) @ Synchronizer.ts:269
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
(anonymous) @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:8
__awaiter @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:4
logSyncSummary @ Synchronizer.ts:246
(anonymous) @ Synchronizer.ts:1104
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
rejected @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:6
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:5
processTicksAndRejections @ node:internal/process/task_queues:95
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:7
(anonymous) @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:8
__awaiter @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/Synchronizer.js:4
start @ Synchronizer.ts:366
(anonymous) @ registry.ts:164
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/registry.js:5
runNextTicks @ node:internal/process/task_queues:60
processTimers @ node:internal/timers:509
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/registry.js:7
fulfilled @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/registry.js:5
runNextTicks @ node:internal/process/task_queues:60
processTimers @ node:internal/timers:509
Promise.then (async)
step @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/registry.js:7
(anonymous) @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/registry.js:8
__awaiter @ /tmp/.mount_Joplinf9Mldk/resources/app.asar/node_modules/@joplin/lib/registry.js:4
```

**Comment by leematos** [comment]
@larntz real s3? Any firewall/proxy/network filtering on your linux machine?

It seems upset about an auth header -- I know there is some failure mode where real S3 can generate a key/secret pair or that doesn't encode properly or some kinda wonky error like that. If it's real s3, do try re-generating a key pair for your linux box and see if that changes anything.

**Comment by larntz** [comment]
@leematos Appreciate the response!

It isn't actually real s3. I should have mentioned that in my original comment. I'm currently using linode object storage, but like I mentioned it is working correctly on my mac and my android phone. 

My laptop does not use a proxy or firewall that would block outgoing connections. I am using the joplin appimage, but I don't think there's anything special that needs configured there for network access (it worked previously when I used minio for storage). 

I can do some more troubleshooting on my side. I can try a real s3 bucket and minio and maybe an older version of joplin.

**Comment by larntz** [comment]
Ok, it is working with s3 and minio. Not sure what is up with linode and why it works on other platforms but not my linux machine. 

I can try to do more testing and see if I can find the solution. I can come back here and update or you can close this out.

**Comment by leematos** [comment]
@larntz I use linode as well, but _not_ with linux. So it's possible that linode is generating a key that _maybe_ has special chars in it?

Looking at where you are dying you are in here:

https://github.com/laurent22/joplin/blob/9c88cb2151fa564a946fa79c4795facf5f9c6594/packages/lib/file-api-driver-amazon-s3.js#L118C1-L131C3

On your linux machine are you able to do the "check config/connection" on s3 setup and is that successful? (I can't remember what it's called off-hand.

OH! Or is your key read-only?

**Comment by larntz** [comment]
It might have been something with it generating a key that had special chars (only special to linux I guess). 

I verified the access key I was using had r/w permission, but created a new one just to see what would happen and IT WORKED!

I don't think I didn't anything weird because I reconfigured it several time and got the same error, but maybe it was me. I'd like to blame the linode key though. :)

I was using the check synchronization button when I was getting the error btw. 

Thanks again for the responses. I should have tried a new access key from the beginning.
